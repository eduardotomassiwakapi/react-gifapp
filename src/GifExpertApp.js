import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = ({defaultCategories = []}) => {

    const [categories, setCategories] = useState(defaultCategories);
    //const [limit, setLimit] = useState(20)
    const limit = 20;
    // const addCategory = (newCat) => {
    //     console.log(categories, newCat);
    //     setCategories( [...categories, newCat] );
    // }

    return (<>
        <h2>- GIFS APP - </h2>
        <AddCategory setCategories={setCategories} />
        {/* <button onClick={ () => addCategory('asd') }>Add</button> */}
        <hr />
        <div >
            {categories.join(' - ')}
        </div>
        <hr />
        <ol>
            {
                categories.map(cat =>
                    <GifGrid key={cat} category={cat} limit={limit} />
                )
            }
        </ol>
    </>)
}

export default GifExpertApp;
