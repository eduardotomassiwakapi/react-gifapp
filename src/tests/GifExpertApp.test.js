import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe('<GifExpertApp />', () => {
    
    // BEFORE TEST
    // let wrapper = shallow(<GifExpertApp />);
    // beforeEach(() => {
    //     wrapper = shallow(<GifExpertApp />);
    // })

    test('should render properly', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should render the correct amount of <GifGrid />', () => {
        const categories = ['Tomatoes', 'Lemons', '']
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        expect(wrapper.find('GifGrid').length).toBe(categories.length)
        expect(wrapper.find('div').text()).toBe(categories.join(' - '))
    })
    
})
