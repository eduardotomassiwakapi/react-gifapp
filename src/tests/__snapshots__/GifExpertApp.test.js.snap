// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifExpertApp /> should render properly 1`] = `
<Fragment>
  <h2>
    - GIFS APP - 
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <div />
  <hr />
  <ol />
</Fragment>
`;
